src/main/scala/nl.knaw.dans.easy.bag2deposit/collections/Collection.scala

```diff
@@ -117,20 +117,29 @@ object Collection extends DebugEnhancedLogging {
     }
 
     trace(skosFile, collectionsFile)
-    for {
-      skosRecords <- parseCsv(skosFile, skosCsvFormat)
-      collectionRecords <- parseCsv(collectionsFile, collectionCsvFormat)
-      originalCollections = collectionRecords.toList.map(parseCollectionRecord)
-      skosMap = skosRecords.map(parseSkosRecord).toMap
-      updatedCollections <- updateCollections(originalCollections)
-    } yield updatedCollections.flatMap { collection =>
-      val name = collection.name
-      lazy val default = <notImplemented>{ s"$name not found in collections skos" }</notImplemented>
-      val elem = skosMap.getOrElse(name, default)
-      collection.members.map(id => id -> elem)
-    }.toMap
-  }.doIfFailure { case e => logger.error(s"could not build CollectionsMap: $cfgDir $e", e) }
-    .getOrElse(Map.empty)
+    val tuples = {
+      for {
+        skosRecords <- parseCsv(skosFile, skosCsvFormat)
+        collectionRecords <- parseCsv(collectionsFile, collectionCsvFormat)
+        originalCollections = collectionRecords.toList.map(parseCollectionRecord)
+        skosMap = skosRecords.map(parseSkosRecord).toMap
+        updatedCollections <- updateCollections(originalCollections)
+      } yield updatedCollections.flatMap { collection =>
+        memberToCollections(skosMap, collection)
+      }
+    }.doIfFailure { case e => logger.error(s"could not build CollectionsMap: $cfgDir $e", e) }
+      .getOrElse(List.empty)
+    tuples.groupBy(_._1).mapValues(_.map(_._2))
+  }
+
+  private def memberToCollections(skosMap: Map[String, Elem], collection: Collection): Seq[(String, Elem)] = {
+    val name = collection.name
+    lazy val default = <notImplemented>
```

Comment:
will this result in a FAILED? i don't want any `<notImplemented>` in the resulting deposits